### PR TITLE
Add a Pill component

### LINKED ISSUE #67: [design-system] Add a Pill component
**Is your feature request related to a problem? Please describe.**
The Pill design is currently seen in Case Triage and Vitals (where it's called "bubble") as well as prospective Practices v2 designs. The implementations are not the same in each place but the end result is.

**Describe the solution you'd like**
Create a Pill component to preclude further redundant implementations. (The "semantic" color scheme from Case Triage is probably not worth preserving, but otherwise that implementation is the most full-featured)

**Describe alternatives you've considered**
Continue to maintain separate implementations. This is probably fine for now but once there is a confirmed additional use case, we should add it to the design system rather than duplicating an existing implementation or creating a third competing one.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

This adopts the Pill component found in Case Triage with a few minor adjustments and should hopefully preclude further redundant implementations.

Here are the following props the Pill component accepts (* required):

- **color*** | sets color for `background-color` if `filled` is true, `border-color` if `filled` is false or is not explicitly defined
 
- **filled** | fills `background-color` when true. True if set explicitly or defined as a prop `<Pill filled={true}>` or `<Pill filled>`, false if set explicitly or undefined as a prop `<Pill filled={false}>` or `<Pill>`

- **textColor** | sets text `color`. If `textColor` is undefined as a prop and `filled` is true, it will default to a light text color. This can be overridden by setting textColor to a color value

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #67

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: packages/design-system/src/components/Pill/Pill.tsx
Comment (macfarlandian) [inline packages/design-system/src/components/Pill/Pill.tsx:28]: probably not necessary to export this

**Review by macfarlandian — APPROVED** [review]
This looks great to me! One minor issue that you can address if you like but doesn't require another review before merging.

File: packages/design-system/src/components/Pill/Pill.tsx
Comment (mxosman) [reply packages/design-system/src/components/Pill/Pill.tsx:28]: Great catch, Ian! Thank you so much for all of your guidance on this one. I will make the adjustment and recommit.

**Pull request merged**
